### ISSUE TITLE
IndexError when splitting data.

### ISSUE THREAD

**Issue by kbruegge** [fact-project/open_crab_sample_analysis#3]
Not sure if this belongs into klaas or not but here goes:

```
kbruegge@phobos:~/open_crab_sample_analysis$ make build/gamma_test.hdf5
klaas_split_data build/gamma_precuts.hdf5 build/gamma \
	-f 0.2 -n train_sep \
	-f 0.2 -n train_reg \
	-f 0.6 -n test \
	-i events
/opt/anaconda3/lib/python3.5/site-packages/h5py/__init__.py:34: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
INFO:root:Found a total of 692836 single-telescope events in the file
INFO:root:Writing 138568 telescope-array events to: build/gamma_train_sep.hdf5
INFO:root:Writing 138568 telescope-array events to: build/gamma_train_reg.hdf5
Traceback (most recent call last):
  File "/opt/anaconda3/lib/python3.5/site-packages/pandas/core/indexing.py", line 1715, in _get_list_axis
    return self.obj.take(key, axis=axis, convert=False)
  File "/opt/anaconda3/lib/python3.5/site-packages/pandas/core/generic.py", line 1928, in take
    convert=True, verify=True)
  File "/opt/anaconda3/lib/python3.5/site-packages/pandas/core/internals.py", line 4002, in take
    indexer = maybe_convert_indices(indexer, n)
  File "/opt/anaconda3/lib/python3.5/site-packages/pandas/core/indexing.py", line 1997, in maybe_convert_indices
    raise IndexError("indices are out-of-bounds")
IndexError: indices are out-of-bounds

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbruegge/.local/bin/klaas_split_data", line 11, in <module>
    load_entry_point('klaas==0.7.0', 'console_scripts', 'klaas_split_data')()
  File "/opt/anaconda3/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/opt/anaconda3/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/opt/anaconda3/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/anaconda3/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/kbruegge/.local/lib/python3.5/site-packages/klaas/scripts/split_data.py", line 117, in main
    data = data.iloc[list(set(data.index.values) - set(selected_data.index))]
  File "/opt/anaconda3/lib/python3.5/site-packages/pandas/core/indexing.py", line 1328, in __getitem__
    return self._getitem_axis(key, axis=0)
  File "/opt/anaconda3/lib/python3.5/site-packages/pandas/core/indexing.py", line 1738, in _getitem_axis
    return self._get_list_axis(key, axis=axis)
  File "/opt/anaconda3/lib/python3.5/site-packages/pandas/core/indexing.py", line 1718, in _get_list_axis
    raise IndexError("positional indexers are out-of-bounds")
IndexError: positional indexers are out-of-bounds
Makefile:75: recipe for target 'build/gamma_test.hdf5' failed
make: *** [build/gamma_test.hdf5] Error 1
kbruegge@phobos:~/open_crab_sample_analysis$ make build/gamma_test.hdf5
klaas_split_data build/gamma_precuts.hdf5 build/gamma \
	-f 0.2 -n train_sep \
	-f 0.2 -n train_reg \
	-f 0.6 -n test \
	-i events
/opt/anaconda3/lib/python3.5/site-packages/h5py/__init__.py:34: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
INFO:root:Found a total of 692836 single-telescope events in the file
INFO:root:Writing 138568 telescope-array events to: build/gamma_train_sep.hdf5
INFO:root:Writing 138568 telescope-array events to: build/gamma_train_reg.hdf5
Traceback (most recent call last):
  File "/opt/anaconda3/lib/python3.5/site-packages/pandas/core/indexing.py", line 1715, in _get_list_axis
    return self.obj.take(key, axis=axis, convert=False)
  File "/opt/anaconda3/lib/python3.5/site-packages/pandas/core/generic.py", line 1928, in take
    convert=True, verify=True)
  File "/opt/anaconda3/lib/python3.5/site-packages/pandas/core/internals.py", line 4002, in take
    indexer = maybe_convert_indices(indexer, n)
  File "/opt/anaconda3/lib/python3.5/site-packages/pandas/core/indexing.py", line 1997, in maybe_convert_indices
    raise IndexError("indices are out-of-bounds")
IndexError: indices are out-of-bounds

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbruegge/.local/bin/klaas_split_data", line 11, in <module>
    load_entry_point('klaas==0.7.0', 'console_scripts', 'klaas_split_data')()
  File "/opt/anaconda3/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/opt/anaconda3/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/opt/anaconda3/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/anaconda3/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/kbruegge/.local/lib/python3.5/site-packages/klaas/scripts/split_data.py", line 117, in main
    data = data.iloc[list(set(data.index.values) - set(selected_data.index))]
  File "/opt/anaconda3/lib/python3.5/site-packages/pandas/core/indexing.py", line 1328, in __getitem__
    return self._getitem_axis(key, axis=0)
  File "/opt/anaconda3/lib/python3.5/site-packages/pandas/core/indexing.py", line 1738, in _getitem_axis
    return self._get_list_axis(key, axis=axis)
  File "/opt/anaconda3/lib/python3.5/site-packages/pandas/core/indexing.py", line 1718, in _get_list_axis
    raise IndexError("positional indexers are out-of-bounds")
IndexError: positional indexers are out-of-bounds
Makefile:75: recipe for target 'build/gamma_test.hdf5' failed
make: *** [build/gamma_test.hdf5] Error 1

```

**Comment by kbruegge** [comment]
I just tried on python 3.6. I get the same error.

**Comment by maxnoe** [comment]
Fixed with using klaas 0.7.3
